tests/integration/suite/test_users.py

```diff
@@ -74,3 +74,53 @@ def check_annotation():
         "clusterscoped.controller.cattle.io/grb-sync_fake"
         not in grb1["metadata"]["finalizers"]
     )
+
+
+def test_roletemplate_finalizer_cleanup(admin_mc, remove_resource):
+    """ This ensures that roletemplates cleanup for clusters < v2.2.8
+        is performed correctly"""
+    client = admin_mc.client
+    rt = client.create_roleTemplate(name="rt-" + random_str())
+    remove_resource(rt)
+    assert rt.annotations["field.cattle.io/rtUpgrade"] == "true"
+
+    # create rt  without rancher api with a bad finalizer
+    api = CustomObjectsApi(admin_mc.k8s_client)
+    json = {
+        "apiVersion": "management.cattle.io/v3",
+        "kind": "RoleTemplate",
+        "metadata": {
+            "finalizers": [
+                "clusterscoped.controller.cattle.io/cluster-roletemplate-sync_fake"
+            ],
+            "name": "test-" + random_str(),
+        }
+    }
+    rt_k8s = api.create_cluster_custom_object(
+        group="management.cattle.io",
+        version="v3",
+        plural="roletemplates",
+        body=json,
+    )
+    remove_resource(rt_k8s["metadata"]["name"])
+    rt_name = rt_k8s["metadata"]["name"]
+    rt_k8s = client.by_id_roleTemplate(id=rt_name)
+
+    def check_annotation():
+        rt1 = client.by_id_roleTemplate(rt_k8s.id)
+        try:
+            if rt1.annotations["field.cattle.io/rtUpgrade"] == "true":
+                return True
+            else:
+                return False
+        except (AttributeError, KeyError):
+            return False
+    wait_for(check_annotation, fail_handler=lambda: "annotation was not added")
+    rt1 = api.get_cluster_custom_object(
+        group="management.cattle.io",
+        version="v3",
+        plural="roletemplates",
+        name=rt_k8s.id,
+    )
+    assert "clusterscoped.controller.cattle.io/grb-sync_fake" \
+        not in rt1["metadata"]["finalizers"]
```

Comment:
Nice test, easy to understand. Spacing seems a little weird but meh.

Reply:
Black formatter + flake8